x-pack/plugins/security_solution/public/cases/components/use_all_cases_modal/index.test.tsx

```diff
@@ -8,16 +8,29 @@
 
 import React from 'react';
 import { renderHook, act } from '@testing-library/react-hooks';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 
 import { useKibana } from '../../../common/lib/kibana';
 import '../../../common/mock/match_media';
-import { TimelineId } from '../../../../common/types/timeline';
 import { useAllCasesModal, UseAllCasesModalProps, UseAllCasesModalReturnedValues } from '.';
 import { TestProviders } from '../../../common/mock';
 
 jest.mock('../../../common/lib/kibana');
+jest.mock('../all_cases', () => {
```

Comment:
is it necessary to mock the component?

Reply:
I think the logic of the modal has nothing to do with the proper function of the `AllCases` component. The tests of the modal are checking the logic of the modal. But tbh, I didn't want to mock all hooks, and API calls the `AllCases` uses just to test if the modal is being closed when `onRowClick` is being called.